packages/tailwindcss/src/compat/plugin-api.ts

```diff
@@ -154,7 +154,7 @@ export function buildPluginApi({
 
       // CSS-in-JS object
       else if (typeof variant === 'object') {
-        designSystem.variants.fromAst(name, objectToAst(variant))
+        designSystem.variants.fromAst(name, objectToAst(variant), designSystem)
```

Comment:
i kinda wish we didn't have to pass the design system into something hanging off the design system :D